### PR TITLE
Add tenants post

### PR DESCRIPTION
**Pull request by mooreds**
This post extolls the benefits of tenants for private labeling.

### CODE REVIEW

**Comment by mooreds** [comment]
Add link to https://fusionauth.io/community/forum/topic/182/basic-questions-regarding-fusionauth when this is posted (re different instances vs tenants).

File: site/_posts/2020-06-05-private-labeling-with-fusionauth.md [hunk 1]
```diff
@@ -1,31 +1,31 @@
 ---
 layout: blog-post
 title: Tenants and private labeling
```
Comment (dpolstra) [inline site/_posts/2020-06-05-private-labeling-with-fusionauth.md]: What if we changed the title to "Tenant APIs and Private Labeling for Multi-tenant Applications"? Does that even work?

Comment (mooreds) [reply site/_posts/2020-06-05-private-labeling-with-fusionauth.md]: How about "How FusionAuth Can Help With Multi-Tenant Private Label Applications"? 

Or "Why You Might Want to Build A Multi-Tenant Private Label App?"

"How You Can Use FusionAuth's Multi-Tenant Feature To Create A Private Label Offering".

Comment (dpolstra) [reply site/_posts/2020-06-05-private-labeling-with-fusionauth.md]: I like "How To Use FusionAuth's Multi-Tenant Feature To Create A Private Label Offering"

File: site/_posts/2020-06-05-private-labeling-with-fusionauth.md [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+
+However, Company1, a large multinational corporation with a big checkbook, approaches you. They want a premium edition; they have decided that your todo application is too good to live without. They want their application to be separate, private labeled, and located at todo.company1.com. They're willing to pay a premium price, as well.
+
+You realize you can offer this easily by modifying your application to respond to multiple hostnames. The logic is pretty straightforward.
+
+But what about your users? The accounts at todo.company1.com should be entirely separate from the accounts at app.todo.com. What if the CEO of Company1 has an account at app.todo.com and signs up for a corporate account? It doesn't make sense to mix personal and business tasks. Further, suppose you sell Organization2 a premium subscription as well. These users should be separate too.
```
Comment (voidmain) [inline site/_posts/2020-06-05-private-labeling-with-fusionauth.md:26]: Should we add in email addresses here to show that his/her account at `app.todo.com` is `alex@company1.com` and his/her account at `todo.company1.com` is also `alex@company1.com`? This is one of the main use-cases for Tenants, when a user needs two accounts that are logically separated but use the same `loginId`.

File: site/_posts/2020-06-05-private-labeling-with-fusionauth.md [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+
+## Tenants to the rescue
+
+FusionAuth provides an easier option: tenants. [Tenants](/docs/v1/tech/core-concepts/tenants) are a first class construct in FusionAuth. When you set up a new instance, there is one tenant installed: "Default". And sometimes one is enough. 
+
+But you can create as many as you'd like. From the perspective of user signing in, each tenant is a separate installation. Each tenant has its own email templates, themes, application configurations and users. API keys can be scoped to a tenant, so if you want to give a client an API key to allow them to create their own integrations, you can. This allows for tighter integrations between your todo application and your clients' systems, and is a nice premium feature to offer with no cost to you.
```
Comment (voidmain) [inline site/_posts/2020-06-05-private-labeling-with-fusionauth.md:50]: "From the perspective of user signing in" should be "From the perspective of a user signing in".

File: site/_posts/2020-06-05-private-labeling-with-fusionauth.md [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+  tenants_response = generic_tenant_client.retrieve_tenants
+  if tenants_response.status != 200
+    puts "Unable to retrieve tenants."
+    return
+  end
+
+  default_tenant_id = tenants_response.success_response.tenants.select { |t| t.name == 'Default' }[0].id
+  default_tenant_theme_id = tenants_response.success_response.tenants.select { |t| t.name == 'Default' }[0].themeId
+
+  new_tenant_request = { "sourceTenantId": default_tenant_id, tenant: {"name": "New client - "+name }}
```
Comment (voidmain) [inline site/_posts/2020-06-05-private-labeling-with-fusionauth.md:84]: Probably need spaces around the `+name` for consistent formatting.

File: site/_posts/2020-06-05-private-labeling-with-fusionauth.md [hunk 5]
```diff
@@ -0,0 +1,107 @@
...
+    puts "Unable to create tenant."
+    puts new_tenant_response.error_response
+    return false
+  end
+  new_tenant = new_tenant_response.success_response.tenant
+  [default_tenant_theme_id, new_tenant]
+end
+```
+
+In this code, we find the tenant with the name `Default` and duplicate it, creating a new tenant with the same settings. We also create a new theme for the new tenant.
```
Comment (voidmain) [inline site/_posts/2020-06-05-private-labeling-with-fusionauth.md:97]: I don't see in the code that it creates a new theme. It looks like it returns a tuple of the default theme id and the new tenant object though.

File: site/_posts/2020-06-05-private-labeling-with-fusionauth.md [hunk 6]
```diff
@@ -0,0 +1,107 @@
...
+end
+```
+
+In this code, we find the tenant with the name `Default` and duplicate it, creating a new tenant with the same settings. We also create a new theme for the new tenant. 
+
+The full [code is available](https://github.com/FusionAuth/fusionauth-example-ruby-tenant-creation) if you want to take a look. This example is written in ruby, but you can use any of our [client libraries](/docs/v1/tech/client-libraries/) to automate this process. 
+
+## Caveats
+
+Once you create a second tenant, API access typically needs to [pass in a tenant identifier](/docs/v1/tech/apis/authentication#making-an-api-request-using-a-tenant-id). This isn't difficult, but you should plan for it.
```
Comment (voidmain) [inline site/_posts/2020-06-05-private-labeling-with-fusionauth.md:103]: Should w show a quick Ruby code snippet of setting the Tenant id when creating the client and retrieving a user or something?

File: site/_posts/2020-06-05-private-labeling-with-fusionauth.md (hunk 5 above)
Comment (mooreds) [reply site/_posts/2020-06-05-private-labeling-with-fusionauth.md]: Whoops, thanks.

File: site/_posts/2020-06-05-private-labeling-with-fusionauth.md (hunk 6 above)
Comment (mooreds) [reply site/_posts/2020-06-05-private-labeling-with-fusionauth.md]: Yup, good idea.

**Pull request merged**